dapps/src/apps/app.rs

```diff
@@ -14,12 +14,10 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-use endpoint::EndpointInfo;
-
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
 pub struct App {
-	pub id: String,
+	pub id: Option<String>,
```

Comment:
Does this have to be unique?

Reply:
Yes, it's used as dapp address (:8545/<id>), but for network dapps we override that with a hash anyway. It's only used as-is for local dapps (you can go to :8545/testdapp)